junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -129,19 +132,18 @@ boolean hasException() {
 
     void notifyException(Exception exception) {
         this.exception = exception;
-        // Commented because of issue https://github.com/k3po/k3po/issues/391
-//        prepared.countDown();
-//        startable.countDown();
-//        finished.countDown();
-        if (testThread != null) {
-            if (testThread != Thread.currentThread())
-                testThread.interrupt();
-        }
+        interruptTestThread();
```

Comment:
Sorry, just saw this comment, which directly speaks to my most recent question.

Wouldn't it make sense to use the non-static thread method `testThread.isInterrupted()` method to determine if the test thread is interrupted (from another thread) without clearing the `interrupted` status?